client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -74,7 +76,11 @@ tie.factory('PythonCodePreprocessorService', [
     // given code snippet. We keep these as instance methods (instead of
     // classmethods) because that makes line number tracking less complicated,
     // since classmethods will need an extra '@classmethod' line.
-    var _wrapCodeIntoClass = function(code, wrapperClassName) {
+    var _wrapCodeIntoClass = function(
+        code, wrapperClassName, shouldProcessInternalFunctionReferences) {
+      if (shouldProcessInternalFunctionReferences) {
+        code = _addClassWrappingToStudentHelperFunctions(code);
```

Comment:
This is better, but still slightly off I think. At this point the function should not need to know whether it's dealing with student code or auxiliary code or whatever -- it should just process the internal function references by prepending the given class name. So the "Student" in the name is not needed and we should be passing in the wrapperClassName.